Modules/IO/NIFTI/test/itkNiftiImageIOTest14.cxx

```diff
@@ -0,0 +1,104 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkNiftiImageIOTest.h"
+#include "itkTestingMacros.h"
+
+// Test writing NIFTI xyzt_units field - should be 10 (mm, sec)
+int
+itkNiftiImageIOTest14(int argc, char * argv[])
+{
+
+  if (argc > 1)
+  {
+    char * testdir = *++argv;
+    itksys::SystemTools::ChangeDirectory(testdir);
+  }
+  else
+  {
+    return EXIT_FAILURE;
+  }
+  int success(0);
+
+  constexpr unsigned int Dimension = 3;
+
+  using PixelType = float;
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  ImageType::Pointer image;
+
+  // fill out an image and write it as nifti.
+  const char * filename = "xyzt_unit_check.nii";
+  using OutputImageType = itk::Image<PixelType, Dimension>;
+  typename OutputImageType::RegionType region;
+  typename OutputImageType::IndexType  start;
+  start[0] = 0;
+  start[1] = 0;
+  start[2] = 0;
```

Comment:
Obviously, if the start index is all zero, it does not need to be set, explicitly. By default, a region starts at index zero.

Reply:
Thanks, I took this from another test. Is it OK if I fix the original also in this PR?

https://github.com/InsightSoftwareConsortium/ITK/blob/7f1c5846087f463ecb7875fb36053096911df92e/Modules/IO/NIFTI/test/itkNiftiImageIOTest5.cxx#L146-L148

Reply:
>  Is it OK if I fix the original also in this PR?

I think so, yes 😃 My preference would be to fix the original in a separate commit. (Rather than squashing everything into one commit.) But it could be part of the same PR, sure. 👍

Reply:
Thanks, I updated with your other feedback, will add a separate commit for the other files

Reply:
Right you are, I removed start also. Just testing now

Reply:
I ended up not using the code to create a blank image, because I added test images to capture more failures. Now I test this by writing one of them back out and checking that the units are correct.